composer.json

```diff
@@ -10,20 +10,29 @@
       "homepage": "http://www.auth0.com/"
     }
   ],
+  "repositories": [
+    {
+      "type": "git",
+      "url": "https://github.com/auth0/php-jwt.git"
+    }
+  ],
   "require": {
-    "php": "^7.1",
-    "guzzlehttp/guzzle": "~6.0 | ~7.0",
+    "php": "^7.3 | ^8.0",
```

Comment:
**How often do we do these changes**
Dropping PHP versions isn't often required, every 2-3 years generally. (Our SDK technically still runs fine on 7.1, but we have to accommodate dependency requirements.)

Moving forward, my goal is for us to follow [the official PHP support schedule](https://www.php.net/supported-versions.php) by deprecating (at year 2) and ultimately dropping (at year 3) support for EOL PHP versions in our SDK, following the official runtime support and security expirations. This would allow us to offer clarity and transparency as to our support windows for customers, and follows best practices in general. It would also have avoided us needing to leapfrog support for two PHP versions at once, as well. But this is a discussion for another day, I suppose.

**Shouldn't these be on a major release instead of a minor?**
Since there are no public API changes on our end, and therefor no breaking changes, this [wouldn't require a major from us according to semver](https://github.com/semver/semver/blob/master/semver.md#what-should-i-do-if-i-update-my-own-dependencies-without-changing-the-public-api).

As our only supported installation method is, like nearly all PHP libraries, through Composer, the version bump won't break users environments as Composer will simply block them from upgrading.

In general, dropping PHP support isn't considered a breaking change in the PHP community. [Here's how Doctrine puts it](https://www.doctrine-project.org/2017/07/25/php-7.1-requirement-and-composer.html);

> One question we frequently hear is, "isn't dropping support for a PHP version a BC break"? In a nutshell, no. A BC break happens when there is an incompatible change that your package manager can't handle. For example, changing a method signature in a minor version is a no-go, since the composer version constraints mentioned above assume any minor upgrade can safely be used.
> 
> However, when we drop support for an older version of PHP, composer will not consider the new version if the PHP version requirement is no longer fulfilled. Thus, you won't end up with a fatal error due to a wrong method signature, you just won't get the new version.

Reply:
I'd treat any **requirement change** as a breaking change that would require a major release if now the conditions are more _restrictive_ than before. 

> As our only supported installation method is, like nearly all PHP libraries, through Composer, the version bump won't break users environments as Composer will simply block them from upgrading.



In this particular case, if you can ensure `Composer` will prevent users from installing this (now incompatible for older PHP setups) new version dependency, then that and the EOL of the previous version is enough to accept this change being released as a minor release.